packages/renewal/sources/renew.move

```diff
@@ -43,8 +43,8 @@ public struct NameRenewed has copy, drop {
 }
 
 /// The renewal's package configuration.
-public struct RenewalConfig has store, drop {
-    config: Config,
+public struct RenewalConfig<phantom T> has store, drop {
```

Comment:
Can we expose the `Renewal<T>` struct from the `suins::pricing` module instead? Also a "new" constructor as well as an immutable access to `PricingConfig`?

That'll make it easier for any higher-level (external) package to provide discounts etc for both registrations and renewals (direct reading from `get_config<Renewal<SUI>>.config().calculate_price()`)